### DISCUSSION TITLE
Joined `hx-target` and `hx-swap` syntax

### DISCUSSION THREAD

**Discussion by imbolc** [bigskysoftware/htmx#2514] [Ideas]
The directives are conceptually very close. They both answer the same question "where to put the response". So why keep them separate?

Considering that `hx-swap` has a static set of values, can't we include them in the name of `hx-target` directive, like:

- `hx-target` - `hx-swap="innerHTML"` for backward compatibility
- `hx-target-outer-html` - `hx-swap="outerHTML"`
- `hx-target-afterbegin` - ...

**Comment by Telroshan** [comment]
Hey, those two attributes define different properties related to the swap behavior ; `hx-target` is indeed "where to put the response", but `hx-swap` is "how".

Some thoughts on this:
- Those 2 properties can be configured independently; you may have any combination of `hx-target` with `hx-swap`, so merging those 2 in a single attribute results in as many different attributes as there are hx-swap values, which implicates more attributes lookups (vs a switch case on the value of a single attribute). In this case I would ask the opposite question; why merge them?

> Considering that `hx-swap` has a static set of values
- `hx-swap` supports additional modifiers (cf [doc](https://htmx.org/attributes/hx-swap/)), making it more complex than simply the initial set of swap methods. Which would imply here having to parse the `hx-target` selector than can be any CSS selector, i.e. potentially a pretty complex one, along `hx-swap` modifiers in the same attribute, making the current parsing way more complex vs the current swap attribute that has a simpler syntax

To be honest, I'm not sure what benefits the suggested approach actually brings, while I'm sure of the challenges and downsides that would come along with it.
